### DISCUSSION TITLE
Few questions.

### DISCUSSION THREAD

**Discussion by Iconology** [open-webui/open-webui#621] [Q&A]
Hey everyone, found ollama a week or two ago, and this today, which is amazing by the way. I have a few questions if you don't mind.

1] Under settings, add-ons, there's an option to set default voice. Is it possible to change the TTS to a different one? I only have the option of default, and not quite sure how or if you can install others.

2) When I click the microphone icon for voice input, it gives an error "Speech recognition error: network."

Thanks everyone

**Comment by justinh-rahb** [comment]
I believe currently all the speech-to-text and text-to-speech are using browser APIs and thus using OS-dependent APIs. On a macOS or Android system, I have many options for these features. On Linux for example, where I don't have any of that stuff setup, nothing works.

**Comment by SmellyBones** [comment]
i was also wondering this, i remember looking into some other tts stuff, this things use the default voices, which you can get more of, but tbh theyer lame but still more options via downloading some through windows options or store i think. i wonder how hard is it to get a voice like Elven labs but working with the ui of ollama. that would be peak llm experience. im okay with the asr it uses, i would prefer whisper. i think its a matter of putting a custom voice file where its pulling its options from.

**Reply by justinh-rahb** [reply]
There's been a lot of ideas thrown around in regards to improving the speech features, including integrating Whisper and/or ElevenLabs. It's on the roadmap but presently there are other features people seem more keen on implementing. PRs are of course always welcome.

**Comment by lorian7392** [comment]
> When I click the microphone icon for voice input, it gives an error "Speech recognition error: network"

I have the same question, but only in the Arc browser. In Chrome, it is normal.
What happened?

**Reply by justinh-rahb** [reply]
Speech API support in different browsers is currently a mess, from what I've gathered recently. This isn't a problem with the WebUI insofar as we're using the standard APIs as they are given and it's just not _great_. The way to solve it would be using or making something custom. I've found at least one other LLM chat app that has gone as far as creating their own package for TTS and STT to work across different platforms consistently.
